.github/workflows/docker.yml

```diff
@@ -1,90 +1,51 @@
-name: Build Docker Images
+name: Build Notebook IDE images
 
 on:
   pull_request:
+    paths:
+      - ".github/workflows/**"
+      - "notebook-servers/**"
+      - "spark/**"
+
   push:
     branches:
       - main
+    paths:
+      - ".github/workflows/**"
+      - "notebook-servers/**"
+      - "spark/**"
```

Comment:
Could you split this into different workflows for each image? i.e. you don't need to build `Spark`, if you change `notebook-server`